src/draw.cpp

```diff
@@ -87,9 +87,44 @@ void draw::set_color(const color_t& c)
 	SDL_SetRenderDrawColor(renderer(), c.r, c.g, c.b, c.a);
 }
 
+/** Some versions of SDL have a bad rectangle drawing implementation. */
+static bool sdl_bad_at_rects()
+{
+	// This could be done once at program start and cached,
+	// but it isn't all that heavy.
+	SDL_version ver;
+	SDL_GetVersion(&ver);
+	if (ver.major > 2 || ver.minor > 0) {
+		return false;
+	}
+	if (ver.patch >= 15 && ver.patch < 18) {
+		return true;
+	}
+	return false;
+}
```

Comment:
This seems like it should use `version_info`?

Reply:
Does that do SDL version handling?

It looks a bit heavy for something low level. Although i guess there's nothing that draws rectangles all that much

Reply:
I mean you give it the major/minor/patch level numbers and then you can compare the objects. At a minimum though, `ver.major > 2 || ver.minor > 0` should surely be `ver.major == 2 || ver.minor == 0`.

Reply:
Actually never mind, turns out I entirely misread it.

Reply:
it not being clear is a pretty good reason to improve it... i'll add a cheap SDL-specific version check. The version_info class does several heap allocations so it's a bit much here.